be/src/formats/parquet/column_reader.h

```diff
@@ -12,13 +12,19 @@ class HdfsScanStats;
 } // namespace starrocks
 
 namespace starrocks::parquet {
+struct ColumnReaderContext {
+    vectorized::Buffer<uint8_t>* selection = nullptr;
+    size_t next_row = 0;
+};
+
```

Comment:
Could you put those files into ColumnReaderOptions. No need for more struct.

Reply:
Another question is, how do you guarantee `selection` is released finally?

Reply:
> Could you put those files into ColumnReaderOptions. No need for more struct.

Now we pass 'const ColumnReaderOptions&' level by level,  if the context members are put into it,  the value can not be update directly.

Reply:
> > Could you put those files into ColumnReaderOptions. No need for more struct.
> 
> Now we pass 'const ColumnReaderOptions&' level by level, if the context members are put into it, the value can not be update directly.

Now the selection (updated to 'filter') points to a local variable, and the object will  be released automatically once the function finish.

Reply:
> Another question is, how do you guarantee `selection` is released finally?

Now the selection (updated to 'filter') points to a local variable, and the object will be released automatically once the function finish.